onnxruntime/core/optimizer/conv_activation_fusion.cc

```diff
@@ -256,43 +259,62 @@ class FuseConvAddRelu : public ReplaceWithNew {
 };
 }  // namespace actions
 
-void RegisterConvActivationFusionRules(SelectorActionRegistry& registry) {
+void RegisterConvActivationFusionRules(SelectorActionRegistry& registry, [[maybe_unused]] bool support_fp16 = false) {
   const auto name = "ConvAct";
   auto action = std::make_unique<actions::FuseConvActivationAction>();
 #if !defined(ORT_MINIMAL_BUILD)
-  auto selector = std::make_unique<selectors::ConvActivationSelector>();
+  auto selector = std::make_unique<selectors::ConvActivationSelector>(support_fp16);
   registry.RegisterSelectorAndAction(name, {{"Conv", {1, 11}}},
                                      std::move(selector), std::move(action));
 #else
   registry.RegisterAction(name, std::move(action));
 #endif
 }
 
-void RegisterConvAddReluFusionRules(SelectorActionRegistry& registry) {
-  const auto name = "ConvAddRelu";
+void RegisterConvAddReluFusionRules(SelectorActionRegistry& registry, [[maybe_unused]] bool support_fp16 = false) {
+  const auto name = "ConvAddReluSelector";
   auto action = std::make_unique<actions::FuseConvAddRelu>();
 #if !defined(ORT_MINIMAL_BUILD)
-  auto selector = std::make_unique<selectors::ConvAddRelu>();
+  auto selector = std::make_unique<selectors::ConvAddReluSelector>(support_fp16);
   registry.RegisterSelectorAndAction(name, {{"Conv", {1, 11}}},
                                      std::move(selector), std::move(action));
 #else
   registry.RegisterAction(name, std::move(action));
 #endif
 }
 
-SelectorActionRegistry CreateSelectorActionRegistry() {
+SelectorActionRegistry CreateSelectorActionRegistry(bool support_fp16 = false) {
   SelectorActionRegistry registry{};
-  RegisterConvActivationFusionRules(registry);
-  RegisterConvAddReluFusionRules(registry);
+  RegisterConvActivationFusionRules(registry, support_fp16);
+  RegisterConvAddReluFusionRules(registry, support_fp16);
   return registry;
 }
 
 }  // namespace
 
 ConvActivationFusion::ConvActivationFusion(const InlinedHashSet<std::string_view>& compatible_execution_providers,
-                                           const SatApplyContextVariant& apply_context)
+                                           std::shared_ptr<KernelRegistry> cpu_kernel_registry,
```

Comment:
I think it would be simpler to pass in a boolean, e.g., `cpu_ep_has_fp16_fused_conv`. that can be computed in graph_transformer_utils.cc.
then we don't need to reset the selector action registry and expose a new member function for that either.

Reply:
I just want to keep it consistent through as nhwcTransformer.

Reply:
ok. there's no need for additional shared ownership here with shared_ptr, it can be a regular pointer.
I think we still don't need UpdateSelectorActionRegistry() - I'd be reluctant to expose more ways to modify something unless necessary. can have a helper function to check whether the kernel registry contains NhwcFusedConv and just pass that result to the first call to CreateSelectorActionRegistry().